src/main/java/com/iexec/worker/compute/post/PostComputeService.java

```diff
@@ -71,8 +71,9 @@ public boolean runStandardPostCompute(TaskDescription taskDescription) {
             return false;
         }
         // encrypt result if needed
-        if (taskDescription.isResultEncryption() && !resultService.encryptResult(chainTaskId)) {
-            log.error("Failed to encrypt result [chainTaskId:{}]", chainTaskId);
+        if (!taskDescription.isTeeTask() && taskDescription.isResultEncryption()) {
```

Comment:
What do you think about moving that check to the beginning?

Reply:
If I'm not wrong we can remove encryptResult method from ResultService

Reply:
You're right! See https://github.com/iExecBlockchainComputing/iexec-worker/pull/431/commits/2b38cfc7d9457da1f1400a4ffdb1915f06c740ee.